### PR TITLE
Add the basic structures of the dependency graph

### PR DESCRIPTION
**Pull request by doom**

### CODE REVIEW

File: libnest/src/cache/depgraph/node.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    #[inline]
+    pub fn dependents_mut(&mut self) -> &mut HashSet<RequirementID> {
+        &mut self.dependents
+    }
+}
+
+impl std::fmt::Display for Node {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match &self.kind {
+            NodeKind::Group { name, .. } => write!(f, "@{}", name.as_str()),
```
Comment (Arignir) [inline libnest/src/cache/depgraph/node.rs]: The `@` shouldn't be needed, no?

File: libnest/src/cache/depgraph/requirement.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+/// The kind of a node's requirement.
+///
+/// A node can hold a requirement to any kind of node: a group, or a package.
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub enum RequirementKind {
+    /// The node requires a group
+    Group {
+        /// The name of the required group
+        name: String,
```
Comment (Arignir) [inline libnest/src/cache/depgraph/requirement.rs]: Shouldn't that be a `GroupName`?

File: libnest/src/cache/depgraph/requirement.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    Package {
+        /// The [`PackageRequirement`] that the package must match.
+        package_req: PackageRequirement,
+    },
+}
+
+impl std::fmt::Display for RequirementKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            RequirementKind::Group { name, .. } => write!(f, "@{}", name),
```
Comment (Arignir) [inline libnest/src/cache/depgraph/requirement.rs]: Same here?

File: libnest/src/cache/depgraph/node.rs (hunk 1 above)
Comment (doom) [reply libnest/src/cache/depgraph/node.rs]: Actually you're right, it's included in the value of `GroupName`s

**Pull request merged**